skxray/speckle_analysis.py

```diff
@@ -0,0 +1,317 @@
+                                labels) for n in range(len(images_set))}
+
+
+def mean_intensity(images, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_intensity : array
+        mean intensity of each ROI for the set of images as an array
+        shape (len(images), number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+
+    index = np.unique(labels)[1:]
+    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+
+    for n, img in enumerate(images):
+        mean_intensity[n] = mean(img, labels, index=index)
+
+    return mean_intensity
+
+
+def combine_mean_intensity(mean_int_dict):
+    """
+    Combine mean intensities of the images(all images sets) for each ROI
+
+    Parameters
+    ----------
+    mean_int_dict : dict
+        mean intensity of each ROI as a dictionary
+        eg: 2 image sets,
+        {image set 1 : (len(images in image set 1), number of labels),
+        image set 2 : (len(images in image set 2), number of labels)}
+
+    Returns
+    -------
+    combine_mean_int : array
+        combine mean intensities of image sets for each ROI of labeled array
+        shape (len(images in all image sets), number of labels)
+
+    """
+    return np.vstack(list(mean_int_dict.values()))
+
+
+def circular_average(image, calibrated_center, threshold=0, nx=100,
+                     pixel_size=None):
+    """
+    Circular average(radial integration) of the intensity distribution of
+    the image data.
+
+    Parameters
+    ----------
+    image : array
+        input image
+
+    calibrated_center : tuple
+        The center in pixels-units (row, col)
+
+    threshold : int, optional
+        threshold value to mask
+
+    nx : int, optional
+        number of bins
+
+    pixel_size : tuple, optional
+        The size of a pixel in real units. (height, width). (mm)
+
+    Returns
+    -------
+    bin_centers : array
+        bin centers from bin edges
+        shape [nx]
+
+    ring_averages : array
+        circular integration of intensity
+    """
+    radial_val = core.radial_grid(calibrated_center, image.shape,
+                                  pixel_size)
+
+    bin_edges, sums, counts = core.bin_1D(np.ravel(radial_val),
+                                          np.ravel(image), nx)
+    th_mask = counts > threshold
+    ring_averages = sums[th_mask] / counts[th_mask]
+
+    bin_centers = core.bin_edges_to_centers(bin_edges)[th_mask]
+
+    return bin_centers, ring_averages
+
+
+def roi_speckle_count_waterfall(images, labels, num=1):
```

Comment:
and can we call this `roi_kymograph` https://en.wikipedia.org/wiki/Kymograph which is the proper name for these kinds of plots.